server/sonar-web/src/main/js/apps/quality-profiles/components/RenameProfileForm.js

```diff
@@ -0,0 +1,132 @@
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+// @flow
+import React from 'react';
+import Modal from 'react-modal';
+import type { Profile } from '../propTypes';
+import { renameProfile } from '../../../api/quality-profiles';
+import { translate, translateWithParameters } from '../../../helpers/l10n';
+
+type Props = {
+  onClose: () => void,
+  onRename: string => void,
+  onRequestFail: Object => void,
+  profile: Profile
+};
+
+type State = {
+  loading: boolean,
+  name: ?string
+};
+
+export default class RenameProfileForm extends React.PureComponent {
+  mounted: boolean;
+  props: Props;
+  state: State = { loading: false, name: null };
+
+  componentDidMount() {
+    this.mounted = true;
+  }
+
+  componentWillUnmount() {
+    this.mounted = false;
+  }
+
+  handleCancelClick = (event: Event) => {
+    event.preventDefault();
+    this.props.onClose();
+  };
+
+  handleNameChange = (event: { currentTarget: HTMLInputElement }) => {
+    this.setState({ name: event.currentTarget.value });
+  };
+
+  handleFormSubmit = (event: Event) => {
+    event.preventDefault();
+
+    const { name } = this.state;
+
+    if (name != null) {
+      this.setState({ loading: true });
+      renameProfile(this.props.profile.key, name).then(
+        () => this.props.onRename(name),
+        error => {
+          if (this.mounted) {
+            this.setState({ loading: false });
+          }
+          this.props.onRequestFail(error);
+        }
+      );
+    }
+  };
+
+  render() {
+    const { profile } = this.props;
+    const header = translateWithParameters(
+      'quality_profiles.rename_x_title',
+      profile.name,
+      profile.languageName
+    );
+
+    return (
+      <Modal
+        isOpen={true}
+        contentLabel={header}
+        className="modal"
+        overlayClassName="modal-overlay"
+        onRequestClose={this.props.onClose}>
+
+        <form id="rename-profile-form" onSubmit={this.handleFormSubmit}>
+          <div className="modal-head">
+            <h2>{header}</h2>
+          </div>
+          <div className="modal-body">
+            <div className="modal-field">
+              <label htmlFor="rename-profile-name">
+                {translate('quality_profiles.new_name')}<em className="mandatory">*</em>
+              </label>
+              <input
+                autoFocus={true}
+                id="rename-profile-name"
+                maxLength="100"
+                name="name"
+                onChange={this.handleNameChange}
+                required={true}
+                size="50"
+                type="text"
+                value={this.state.name != null ? this.state.name : profile.name}
+              />
+            </div>
+          </div>
+          <div className="modal-foot">
+            {this.state.loading && <i className="spinner spacer-right" />}
+            <button disabled={this.state.loading || !this.state.name} id="rename-profile-submit">
```

Comment:
Like for the `copy form`, I think it would be good to not allow submit if the name didn't change